### DISCUSSION TITLE
Quarkus Lambda Rest extension with other @Named handlers

### DISCUSSION THREAD

**Discussion by sivabalachandran** [quarkusio/quarkus#35327] [Q&A]
I am writing a Lambda that has a REST handler and a handler that listens to SNS event. I want two handlers to be in same repo as they share majority of the code and does pretty much the same thing. Currently this does not work with exception `Multiple handler classes.  You have a custom handler class....`. I see the rest extension creating a **`ProvidedAmazonLambdaHandlerBuildItem`** that basically leads to this issue. I am wondering if we can do a sensible default in the extension by creating the build item conditionally if no handler is configured in **`application.properties`**.

I can do a pull request, but I am not sure how do I resolve config values at build time? Need some direction here. Thanks.

[a comment by an automated account is not shown]

**Comment by cthiebault** [comment]
Did you find a solution for this @sivabalachandran?

I'm considering to split my project in 3 sub-modules to support this but it's not very elegant :-(
- `commons` (contains the business logic called by the handlers)
- `lambda-rest` (contains only the REST lambda using `quarkus-amazon-lambda-rest`)
- `lambda-sns` (contains all other named handlers)

And as we use Quarkus, the `commons` module would need to be packaged as a Quarkus Extension :-(

**Comment by tortru** [comment]
A little bit late but maybe someone else has the same question.

You do not need to make your commons a Quarkus extension. It can be a Maven dependency.

For such projects I use the Maven multi-module project approach.
In your case lambda-rest and lambda-sns would be defined as modules in your parent POM and both have commons as a dependency which is managed in the dependencyManagement section of the parent POM.

The problem is if you want to build a native executable of your application this increases the build time because each handler is built separately.
